x/ibc/04-channel/keeper/packet.go

```diff
@@ -462,7 +464,9 @@ func (k Keeper) AcknowledgementExecuted(
 
 		nextSequenceAck++
 
-		k.SetNextSequenceAck(ctx, packet.GetDestPort(), packet.GetDestChannel(), nextSequenceAck)
+		// incrementng NextSequenceAck and storing under this chain's channelEnd identifiers
+		// Since this is the original sending chain, our channelEnd is packet's source port and channel
+		k.SetNextSequenceAck(ctx, packet.GetSourcePort(), packet.GetSourceChannel(), nextSequenceAck)
```

Comment:
main diff

Reply:
love the comments :heart: :heart: :heart: